lib/msf/core/post_mixin.rb

```diff
@@ -46,9 +49,9 @@ def setup
     # for its platform, capabilities, etc.
     check_for_session_readiness if session.type == "meterpreter"
 
-    incompatibility = session_incompatibility(session)
-    unless incompatibility.nil?
-      print_warning("SESSION may not be compatible with this module (#{incompatibility})")
+    incompatibility_reasons = session_incompatibility_reasons(session)
+    unless incompatibility_reasons.empty?
+      print_warning("SESSION may not be compatible with this module (#{incompatibility_reasons.join('. ')})")
```

Comment:
Since we're getting more and more verbose, would it make sense to print each reason on a new line?

Something like the following might make it a little more clear that there are multiple problems.

```
[!] SESSION may not be compatible with this module:
[!]   * incompatible session platform: linux
[!]   * missing Meterpreter features: stdapi_ui_desktop_screenshot, stdapi_ui_send_keyevent, stdapi_ui_send_mouse, espia_image_get_dev_screen
```

Reply:
Works for me, updated :+1:

```
msf6 post(multi/manage/screenshare) > run session=-1
[!] SESSION may not be compatible with this module:
[!] 	 * missing Meterpreter features: stdapi_ui_desktop_screenshot, stdapi_ui_send_keyevent, stdapi_ui_send_mouse, espia_image_get_dev_screen
```